### PR TITLE
feat(issuer): [SFEQS-1351] Require at least one mandatory signature

### PR DESCRIPTION
**Pull request by lucacavallaro**
#### List of Changes

1. Created a new codec for `SignatureFields` that checks the "require at least one mandatory signature" constraint

#### How Has This Been Tested?

The change has been unit tested

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/io-sign/src/document.ts [hunk 1]
```diff
@@ -90,9 +90,23 @@ export const PdfDocumentMetadata = t.type({
...
+export interface SignatureFieldsBrand {
+  readonly SignatureFields: unique symbol;
+}
+
+export const SignatureFields = t.brand(
+  t.array(SignatureField),
+  (
+    signatureFields
+  ): signatureFields is t.Branded<SignatureField[], SignatureFieldsBrand> =>
+    signatureFields.length === 0 ||
```
Comment (grausof) [inline packages/io-sign/src/document.ts:102]: It seems to me that if the list of signature fields is empty, however, we can proceed with the signature on the entire file!

File: packages/io-sign/src/__test__/document.spec.ts [hunk 2]
```diff
@@ -94,3 +101,55 @@ describe("Document", () => {
...
+        attributes: {
+          uniqueName: title as NonEmptyString,
+        },
+        clause: {
+          title: title as WithinRangeString<5, 80>,
+          type,
+        },
+      };
+    };
+    const testTable: Array<{
```
Comment (gunzip) [inline packages/io-sign/src/__test__/document.spec.ts:125]: you probably can do this using `it.each`

**Review by gunzip — APPROVED** [review]

File: packages/io-sign/src/document.ts (hunk 1 above)
Comment (lucacavallaro) [reply packages/io-sign/src/document.ts:102]: yes, this check ensures that!

the signature field list must be empty or with at least one mandatory signature

Comment (grausof) [reply packages/io-sign/src/document.ts:102]: OK!

**Review by grausof — APPROVED** [review]
LGTM

File: packages/io-sign/src/__test__/document.spec.ts (hunk 2 above)
Comment (lucacavallaro) [reply packages/io-sign/src/__test__/document.spec.ts:125]: I tried to write with `it.each`, I confirm that it makes add types much more difficult

**Pull request merged**